execute/tokendata/usdc/cctpv2/http_client.go

```diff
@@ -0,0 +1,186 @@
+// Exposes an HTTP client to call the Circle Cross-Chain Transfer Protocol (CCTP) V2 API.
+//
+// The goal is to fetch attestations for USDC transfers made via CCTPv2 by CCIP. The /v2/messages/ endpoint
+// takes a source domain ID and transaction hash and returns a list of CCTPv2 messages executed on that source and
+// transaction. Each message contains an attestation and a decoded message body, which are needed to pass to
+// destination chain contracts where the attestation will be verified and the USDC transfer will be executed (e.g. the
+// USDC will be minted and transferred on the destination chain).
+//
+// API docs are here: https://developers.circle.com/api-reference/stablecoins/common/get-messages-v-2
+//
+// An example API calls is:
+//nolint:lll
+// https://iris-api-sandbox.circle.com/v2/messages/0?transactionHash=0x8f1d58a2161b0bd662609b62537d57b76cfa6bb22a9afe8ec49b09f3810d44c2
+
+package cctpv2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+	"github.com/smartcontractkit/chainlink-ccip/execute/tokendata/http"
+	"github.com/smartcontractkit/chainlink-ccip/execute/tokendata/usdc"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+const (
+	apiVersionV2              = "v2"
+	messagesPath              = "messages"
+	attestationStatusComplete = "complete"
+)
+
+type CTTPv2AttestationClient struct {
+	lggr   logger.Logger
+	client http.HTTPClient
+}
+
+func NewCTTPv2AttestationClient(
+	lggr logger.Logger,
+	config pluginconfig.USDCCCTPObserverConfig,
+) (*CTTPv2AttestationClient, error) {
+	client, err := http.GetHTTPClient(
+		lggr,
+		config.AttestationAPI,
+		config.AttestationAPIInterval.Duration(),
```

Comment:
How does the rate limiting work when both v1 and v2 are in use? I mean, is there a shared rate limit state between v1 and v2? Or is the rate limitting per api version? `http.GetHTTPClient` enforces a single instance (with shared rate limit) per `attestation api` with version exluded

Reply:
I assume the rate limiting is shared across v1 and v2